UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.API.FunctionalTests/FunctionalTests/D365HttpPayloadValidation.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using NUnit.Framework;
+using UKHO.ExternalNotificationService.API.FunctionalTests.Helper;
+using UKHO.ExternalNotificationService.API.FunctionalTests.Model;
+
+namespace UKHO.ExternalNotificationService.API.FunctionalTests.FunctionalTests
+{
+    class D365HttpPayloadValidation
+    {
+        private EnsApiClient _ensApiClient { get; set; }
+        private TestConfiguration _testConfig { get; set; }
+        private D365Payload _d365Payload { get; set; }
+
+
+        [SetUp]
+        public void Setup()
+        {
+            _testConfig = new TestConfiguration();
+            _ensApiClient = new EnsApiClient(_testConfig.EnsApiBaseUrl);
+
+            string filePath = Path.Combine(Directory.GetCurrentDirectory(), _testConfig.PayloadFolder, _testConfig.PayloadFileName);
+
+            _d365Payload = JsonConvert.DeserializeObject<D365Payload>(File.ReadAllText(filePath));
+
+        }
+
+        [Test]
+        public async Task WhenICallTheEnsSubscriptionApiWithAValidD365Payload_ThenAcceptedStatusIsReturned()
+        {
+             
+            var apiResponse = await _ensApiClient.PostEnsApiSubcriptionAsync(_d365Payload);
+            Assert.AreEqual(202, (int)apiResponse.StatusCode, $"Incorrect status code {apiResponse.StatusCode}  is  returned, instead of the expected 202.");
+
+        }
+
+        [Test]
+        public async Task WhenICallTheEnsSubscriptionApiWithEmptyD365Payload_ThenABadRequestStatusIsReturned()
+        {
+            D365Payload d365PayloadEmpty = null;
+
+            var apiResponse = await _ensApiClient.PostEnsApiSubcriptionAsync(d365PayloadEmpty);
```

Comment:
May as well just use `null` directly here I think:
```suggestion
            var apiResponse = await _ensApiClient.PostEnsApiSubcriptionAsync(null);
```

Reply:
completed required changes.